File: 75 days Challenge Leetcode/Queue/Day-27/Solution.py
from collections import deque

class RecentCounter:
    def __init__(self):
        self.Q = deque()  # Initialize the deque to store ping timestamps

    def ping(self, t: int) -> int:
        self.Q.append(t)  # Add the current ping timestamp

        # Remove timestamps that are older than 3000 ms from the current ping
        while self.Q and self.Q[0] < t - 3000:
            self.Q.popleft()

        return len(self.Q)  # Return the number of pings in the last 3000 ms

File: 75 days Challenge Leetcode/Queue/Day-27/Solution.cpp
#include <queue>

class RecentCounter {
public:
    queue<int> Q; // Queue to store the ping timestamps

    RecentCounter() {
        // Constructor
    }

    int ping(int t) {
        Q.push(t); // Add the current ping timestamp

        // Remove timestamps that are older than 3000 ms from the current ping
        while (!Q.empty() && Q.front() < t - 3000) 
            Q.pop();

        return Q.size(); // Return the number of pings in the last 3000 ms
    }
};


/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */

